include/ofi_shm.h

```diff
@@ -193,6 +193,15 @@ struct smr_map {
 	struct smr_peer	peers[SMR_MAX_PEERS];
 };
 
+struct smr_queue_pool_offsets {
+	size_t		cmd_queue_offset;
+	size_t		resp_queue_offset;
+	size_t		inject_pool_offset;
+	size_t		sar_pool_offset;
+	size_t		peer_data_offset;
+	size_t		name_offset;
+};
+
```

Comment:
I don't love defining this struct. Personally, I would rather have a function that takes in the tx_size and rx_size and pointers to the offsets to set them in one place. But I will also bow down to whatever the all mighty @shefty has to say about it.

Reply:
I thought about having the function `smr_shm_size_per_ep()` which takes in tx/rx_size, and a pointer to struct smr_region (*smr). But the problem is that the *smr has not been mmap'ed when calling `smr_shm_size_per_ep()` call, so we are not able to access those offsets of struct smr_region in the call to `smr_shm_size_per_ep()`.  The following code sample briefly shows the problem.
```
int smr_create(const struct fi_provider *prov, struct smr_map *map,
               const struct smr_attr *attr, struct smr_region **smr)
{
          ...
          total_size = smr_shm_size_per_ep(tx_size, rx_size, *smr);
          map_addr = shm_open/ftruncate/mmap(total_size);
          *smr = mmap_addr;
          ...
}
```

@shefty @aingerson I'd love to change if you have any better idea.

Reply:
I was thinking of passing in the offsets themselves and maybe returning the total size like this:
```
size_t inline smr_calculate_offsets(size_t tx_size, size_t rx_size, size_t *cmd_offset,
                   size _t *resp_offset, size_t *inject_offset, size_t *sar_offset,
                   size_t *peer_offset, size_t *name_offset)
```
In the smr_create, we could just pass in the variables as they are and in the size calculation, you could just just the return. You could even have ifs in the function to be able to handle NULL parameters and only set them if supplied.

Reply:
Thank you @aingerson, I see your point. I was a bit concerned about pushing the number of arguments. But it looks not that many actually, I will proceed in this way.